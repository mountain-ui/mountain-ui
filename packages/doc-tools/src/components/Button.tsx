import React from "react";

export interface ButtonProps extends React.ComponentProps<'button'> { };

export const Button = ({ className, ...props }: ButtonProps) => {
  const classes = [
    'relative inline-block w-fit border-0 rounded-lg font-medium text-body py-2 px-4 cursor-pointer transition ease duration-150 disabled:text-gray-700 disabled:bg-gray-100 disabled:cursor-not-allowed',
    'bg-opacity-20 hover:bg-opacity-30 text-blue-700 bg-blue-600 from-blue-600 to-blue-400',
    className
  ].filter(Boolean).join(' ')

  return <button className={classes} {...props} />
}

Button.displayName = 'Button';
