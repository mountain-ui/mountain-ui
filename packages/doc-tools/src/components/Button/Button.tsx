import styled from 'styled-components';

import { BaseElementProps, styleProps } from '../BaseElement';

export type ButtonProps = BaseElementProps;

/**
 * Primary UI component for user interaction
 */
const Button = styled.button<ButtonProps>`
  position: relative;
  background-color: hsl(206, 84%, 60%);
  border: 0;
  border-radius: 12px;
  color: #ffffff;
  font-weight: bold;
  font-size: 18px;
  padding: 8px;
  cursor: pointer;
  text-shadow: 1px 1px 6px rgb(0 0 0 / 20%);
  transition: all 0.3s;

  &:disabled,
  &:disabled:hover {
    color: gray;
    background-color: #f0f0f0;
    cursor: not-allowed;
  }
  ${styleProps}
`;

Button.displayName = 'Button';

export default Button;
