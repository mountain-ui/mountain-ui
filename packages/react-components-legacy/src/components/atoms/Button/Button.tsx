import css from '@styled-system/css';
import styled from 'styled-components';
import { variant } from 'styled-system';

import { BaseElementProps, styleProps } from '../../BaseElement';

export type ButtonProps = BaseElementProps & {
  /** Button variant */
  variant?: 'primary' | 'outline' | 'success' | 'warning' | 'error';
};

/**
 * Primary UI component for user interaction
 */
const Button = styled.button<ButtonProps>`
  ${css({
    position: 'relative',
    border: 0,
    borderRadius: 3,
    boxShadow: 1,
    color: 'white',
    fontWeight: 'semiBold',
    fontSize: 2,
    padding: 3,
    cursor: 'pointer',
    textShadow: '1px 1px 6px rgb(0 0 0 / 20%)',
    transition: 'all .3s ease',
    ':hover': {
      boxShadow: 2
    },
    ':disabled, :disabled:hover': {
      color: 'gray.700',
      backgroundColor: 'gray.100',
      cursor: 'not-allowed',
      boxShadow: 1
    }
  })}
  ${variant({
    scale: 'variants.button',
    variants: {
      primary: {
        backgroundColor: 'blue.400'
      },
      outline: {
        backgroundColor: 'transparent',
        boxShadow: 'inset 0 0 0 3px;',
        color: 'blue.400',
        ':hover': {
          color: 'white'
        },
        '::before, :hover::before': {
          content: '""',
          position: 'absolute',
          top: 0,
          left: 0,
          height: '100%',
          width: '100%',
          transitionDuration: '0.25s',
          boxShadow: 'inset 0 0 0 3px',
          borderRadius: 3,
          color: 'blue.400',
          zIndex: -1
        },
        ':hover::before': {
          boxShadow: 'inset 0 0 0 30px',
          borderRadius: 3
        }
      },
      success: {
        backgroundColor: 'green.400'
      },
      warning: {
        backgroundColor: 'yellow.400'
      },
      error: {
        backgroundColor: 'red.400'
      }
    }
  })}
  ${styleProps}
`;

Button.defaultProps = {
  variant: 'primary'
};

Button.displayName = 'Button';

export default Button;
