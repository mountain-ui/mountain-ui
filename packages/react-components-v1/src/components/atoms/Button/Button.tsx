import { createMntComponent } from '../../../internals/create_mnt_component';

type ButtonVariant = 'primary' | 'outline' | 'success' | 'warning' | 'error'

export interface ButtonProps { }


const baseClass = 'relative border-0 rounded-lg shadow text-light font-medium text-body p-2 cursor-pointer drop-shadow-md transition ease duration-300 hover:shadow-md disabled:text-gray-700 disabled:bg-gray-100 disabled:cursor-not-allowed';
const variants: Record<ButtonVariant, string> = {
  primary: 'bg-blue-400',
  outline: '',
  success: 'bg-green-400',
  warning: 'bg-yellow-400',
  error: 'bg-red-400',
}


/**
 * Primary UI component for user interaction
 */
const Button = createMntComponent<ButtonProps>('button')({
  baseClass,
  variants,
  defaultVariant: 'primary',
})

Button.displayName = 'Button';

export default Button;
