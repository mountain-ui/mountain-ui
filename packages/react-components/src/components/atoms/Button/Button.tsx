import { mnt } from 'mnt-internals';

export const BUTTON_BASE_CLASS = 'relative inline-block w-fit border-0 rounded-lg font-medium text-body py-2 px-4 cursor-pointer transition ease duration-150 disabled:text-gray-700 disabled:bg-gray-100 disabled:cursor-not-allowed';
export const BUTTON_COLORS = {
  primary: 'text-blue-700 bg-blue-600 from-blue-600 to-blue-400',
  accent: 'text-purple-700 bg-purple-600 from-purple-600 to-purple-400',
  success: 'text-green-600 bg-green-600 from-green-600 to-green-400',
  warning: 'text-yellow-600 bg-yellow-500 from-yellow-600 to-yellow-400',
  danger: 'text-red-700 bg-red-600 from-red-600 to-red-400',
  disabled: 'text-gray-600 bg-gray-500 from-gray-600 to-gray-400'
} as const;
export const BUTTON_VARIANTS = {
  shade: 'bg-opacity-20 hover:bg-opacity-30',
  solid: '!text-light',
  outline: 'bg-opacity-10 hover:bg-opacity-30 shadow-outline-button',
  gradient: 'bg-gradient-to-br hover:bg-gradient-to-r !text-light',
  link: '!bg-transparent hover:underline underline-offset-4',
  text: '!bg-transparent hover:scale-105',
} as const;

export type MntButtonColor = keyof typeof BUTTON_COLORS
export type MntButtonVariant = keyof typeof BUTTON_VARIANTS

export interface MntButtonProps {
  /**
   * The color to apply for the button content
   */
  color?: MntButtonColor
  /**
   * The button status
   */
  disabled?: boolean
  /**
   * The variant to use for the button
   */
  variant?: MntButtonVariant
}

/**
 * Button component for triggering actions or events in the UI. 
 * Provides interactivity with styled visuals, enhancing user engagement and navigation within the application.
 */
export const MntButton = mnt('button') <MntButtonProps>`
  ${BUTTON_BASE_CLASS}
  ${({ color = 'primary' }) => BUTTON_COLORS[color]}
  ${({ variant = 'shade' }) => BUTTON_VARIANTS[variant]}
`

MntButton.displayName = 'MntButton';