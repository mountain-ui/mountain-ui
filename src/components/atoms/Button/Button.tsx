import css from '@styled-system/css';
import styled from 'styled-components'
import { variant } from 'styled-system';

import { BaseElement, BaseElementProps } from '../../BaseElement';

export type ButtonProps = BaseElementProps & {
  /**
   * Button variant
   */
  variant: 'primary' | 'outline' | 'success' | 'error'
}

/**
 * Primary UI component for user interaction
 */
const Button = styled(BaseElement)<ButtonProps>(
  css({
    color: 'text.light',
    fontWeight: 'ultraBold',
    textShadow: '1px 1px 6px rgb(0 0 0 / 20%)',
    padding: 3,
    borderRadius: 3,
    border: 0,
    boxShadow: 1,
    transition: 'box-shadow .3s ease',
    ':hover': {
      boxShadow: 2,
    }
  }),
  variant({
    scale: 'variants.button',
    variants: {
      success: {
        background: 'linear-gradient(30deg, rgba(38,228,95,1) 0%, rgba(118,241,156,1) 60%, rgba(171,246,193,1) 100%)',
      },
      error: {
        background: 'linear-gradient(30deg, rgba(247,77,77,1) 0%, rgba(251,97,97,1) 60%, rgba(255,124,124,1) 100%)',
      },
      outline: {
        color: 'text.primary',
        backgroundColor: 'transparent',
        padding: 3,
        border: '2px solid',
        borderColor: 'text.primary',
        borderRadius: 3,
        boxShadow: 0
      },
    }
  })
)

Button.defaultProps = {
  variant: 'primary',
  as: 'button'
}

Button.displayName = 'Button'

export default Button;