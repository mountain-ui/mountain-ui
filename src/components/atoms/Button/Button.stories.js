import React from 'react';

import Button from './Button';

export default {
  title: 'Atoms/Button',
  component: Button
};

export const Primary = args => <Button {...args}>Click</Button>;

Primary.args = {
  variant: 'success'
};

export const Outline = args => <Button {...args}>Click</Button>;

Outline.args = {
  variant: 'outline'
};

Primary.parameters = Outline.parameters = {
  jest: ['Button.test.js']
};
