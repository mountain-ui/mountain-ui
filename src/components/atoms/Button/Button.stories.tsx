import React from 'react'
import { Meta, Story } from '@storybook/react/types-6-0';

import Button, { ButtonProps } from './Button';

export default {
  title: 'Atoms/Button',
  component: Button,
  argTypes: {
    backgroundColor: { control: 'color' }
  }
} as Meta;

// const Template: Story<ButtonProps> = args => <Button {...args} />;

export const Basic: Story<ButtonProps> = (args) => <Button {...args}>Click</Button>;

Basic.args = {
  backgroundColor: 'red'
};

Basic.parameters = {
  jest: ['Button.test.js'],
};