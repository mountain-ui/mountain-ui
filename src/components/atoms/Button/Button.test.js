import React from 'react';
import initStoryshots from '@storybook/addon-storyshots';
import { render } from '@testing-library/react';

import { Outline, Primary } from './Button.stories';

initStoryshots();

describe('Button', () => {
  describe('primary variant', () => {
    it('should render correctly on mount', () => {
      const { container } = render(<Primary {...Primary.args} />);
      expect(container).toBeInTheDocument();
    });
  });

  describe('outline variant', () => {
    it('should render correctly on mount', () => {
      const { container } = render(<Outline {...Outline.args} />);
      expect(container).toBeInTheDocument();
    });
  });
});
