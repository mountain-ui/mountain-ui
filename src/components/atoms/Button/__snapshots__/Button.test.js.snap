// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Button Outline 1`] = `
<div
  className="sc-bdfBwQ sc-dlfnbm sc-hKgILt QQPef dtVUxq"
>
  <button
    className="sc-bdfBwQ sc-gsTCUz  BanWB"
  >
    Click
  </button>
</div>
`;

exports[`Storyshots Atoms/Button Primary 1`] = `
<div
  className="sc-bdfBwQ sc-dlfnbm sc-hKgILt QQPef dtVUxq"
>
  <button
    className="sc-bdfBwQ sc-gsTCUz  kqSVpZ"
  >
    Click
  </button>
</div>
`;

exports[`Storyshots Atoms/Layout/Container Container 1`] = `
<div
  className="sc-bdfBwQ sc-dlfnbm sc-hKgILt QQPef dtVUxq"
>
  <div
    className="sc-bdfBwQ sc-dlfnbm sc-hKgILt gyNwPp dtVUxq"
    height="100vh"
  >
    This is the content of a container component
  </div>
</div>
`;

exports[`Storyshots Atoms/Layout/Grid Grid 1`] = `
<div
  className="sc-bdfBwQ sc-dlfnbm sc-hKgILt QQPef dtVUxq"
>
  <div
    className="sc-bdfBwQ sc-dlfnbm sc-eCssSg  dtVUxq jhIYme"
  >
    <div
      className="sc-bdfBwQ sc-dlfnbm sc-eCssSg  dtVUxq ehdhyK"
    >
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
    </div>
    <div
      className="sc-bdfBwQ sc-dlfnbm sc-eCssSg  dtVUxq lllEAR"
    >
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
    </div>
    <div
      className="sc-bdfBwQ sc-dlfnbm sc-eCssSg  dtVUxq cEGeNL"
    >
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
      <div
        className="sc-bdfBwQ sc-dlfnbm hBRVxd dtVUxq"
        height={150}
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots Atoms/Typography/Heading Heading 1`] = `
<div
  className="sc-bdfBwQ sc-dlfnbm sc-hKgILt QQPef dtVUxq"
>
  <h1
    className="sc-bdfBwQ sc-jSgupP sc-gKsewC gYDVUp jJYlhf kWykmG"
    color="text.primary"
  >
    Mountain UI library
  </h1>
</div>
`;

exports[`Storyshots Atoms/Typography/Paragraph Paragraph 1`] = `
<div
  className="sc-bdfBwQ sc-dlfnbm sc-hKgILt QQPef dtVUxq"
>
  <p
    className="sc-bdfBwQ sc-jSgupP sc-iBPRYJ sc-fubCfw  jJYlhf"
  >
    Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
  </p>
</div>
`;

exports[`Storyshots Atoms/Typography/Text Text 1`] = `
<div
  className="sc-bdfBwQ sc-dlfnbm sc-hKgILt QQPef dtVUxq"
>
  <span
    className="sc-bdfBwQ sc-jSgupP sc-iBPRYJ  jJYlhf"
  >
    Mountain UI library
  </span>
</div>
`;
